src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -0,0 +1,5392 @@
+/**
+ * (C) Copyright 2022-2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+
+#define D_LOGFAC     DD_FAC(il)
+
+#include <stdio.h>
+#include <dirent.h>
+#include <dlfcn.h>
+#include <sys/types.h>
+#include <utime.h>
+#include <sys/stat.h>
+#include <sys/vfs.h>
+#include <fcntl.h>
+#include <stdarg.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/statfs.h>
+#include <sys/statvfs.h>
+#include <pthread.h>
+#include <sys/ioctl.h>
+#include <mntent.h>
+#include <signal.h>
+#include <inttypes.h>
+
+#include <daos/debug.h>
+#include <gurt/list.h>
+#include <gurt/common.h>
+#include <gurt/hash.h>
+#include <daos.h>
+#include <daos_fs.h>
+#include <daos_uns.h>
+#include <dfuse_ioctl.h>
+#include <daos_prop.h>
+#include <daos/common.h>
+#include "dfs_internal.h"
+
+#include "hook.h"
+
+/* D_ALLOC and D_FREE can not be used in query_path(). It causes dead lock during daos_init(). */
+#define FREE(ptr)	({if ((ptr) != NULL) free(ptr); })
+
+/* Use very large synthetic FD to distinguish regular FD from Kernel */
+
+/* FD_FILE_BASE - The base number of the file descriptor for a regular file.
+ * The fd allocate from this lib is always larger than FD_FILE_BASE.
+ */
+#define FD_FILE_BASE        (0x20000000)
+
+/* FD_FILE_BASE - The base number of the file descriptor for a directory.
+ * The fd allocate from this lib is always larger than FD_FILE_BASE.
+ */
+#define FD_DIR_BASE         (0x40000000)
+
+/* The max number of mount points for DAOS mounted simultaneously */
+#define MAX_DAOS_MT         (8)
+
+#define READ_DIR_BATCH_SIZE (96)
+#define MAX_FD_DUP2ED       (8)
+
+#define MAX_MMAP_BLOCK      (64)
+
+#define MAX_OPENED_FILE     (2048)
+#define MAX_OPENED_FILE_M1  ((MAX_OPENED_FILE)-1)
+#define MAX_OPENED_DIR      (512)
+#define MAX_OPENED_DIR_M1   ((MAX_OPENED_DIR)-1)
+
+/* The buffer size used for reading/writing in rename() */
+#define FILE_BUFFER_SIZE    (64 * 1024 * 1024)
+
+/* Create a fake st_ino in stat for a path */
+#define FAKE_ST_INO(path)   (d_hash_string_u32(path, strnlen(path, DFS_MAX_PATH)))
```

Comment:
Thank you for your suggestion! I checked the code that call FAKE_ST_INO(). The length of "path" is not used around. I would prefer staying current form at this time. I will keep your suggestion in mind in case the situation is changed in future.